src/indexers/offset_tracker/rated.py

```diff
@@ -0,0 +1,42 @@
+from datetime import UTC, datetime
+from pydantic import StrictStr
+
+from src.clients.slaos import SlaosClient
+from src.config.models.offset import OffsetYamlConfig, OffsetSlaosYamlConfig
+from src.config.models.output import RatedOutputConfig
+from src.indexers.offset_tracker.base import OffsetTracker
+
+
+class RatedAPIOffsetTracker(OffsetTracker):
+    _current_offset: int
+
+    def __init__(self, config: OffsetYamlConfig, integration_prefix: StrictStr):
+        super().__init__(config=config, integration_prefix=integration_prefix)
+        slaos_config: OffsetSlaosYamlConfig = config.slaos  # type: ignore[assignment]
+        client_config = RatedOutputConfig(**slaos_config.model_dump())
+        self.client = SlaosClient(client_config)
+        self.initialise_offset()
+
+    def get_current_offset(self) -> int:  # type: ignore[return-value]
+        return self._current_offset
+
+    def update_offset(self, offset: int) -> None:
+        # Just keep track of offset internally, no need to send it to the API.
+        self._current_offset = offset
+
+    def initialise_offset(self) -> None:
+        if self.config.override_start_from:
+            offset = self.config.start_from
+        else:
+            api_timestamp = self.get_offset_from_api()
+            if api_timestamp is not None:
+                from_epoch = api_timestamp - datetime(1970, 1, 1, tzinfo=UTC)
+                offset = int(from_epoch.total_seconds())
+            else:
+                offset = self.config.start_from
```

Comment:
These offsets are different. One is seconds from epoch, the other is milliseconds.